share/doc/wake/wake-for-fsharp-developers.md

```diff
@@ -0,0 +1,150 @@
+# Wake for F# developers
+
+F# and Wake are both ML-dialects, so they share very similar heritages as far as programming languages go. The major difference is that F# is intended to be a general-purpose programming language and is multi-paradigm since it also supports imperative and OOP programming, whereas Wake is intended to be a programming language to support a build system and does not support the imperative or OOP programming paradigms. However, while simple, the Wake programming language is rather powerful.
```

Comment:
Slight tweak to avoid suggesting Wake's not sufficiently powerful.
```suggestion
F# and Wake are both ML-dialects, so they share very similar heritages as far as programming languages go. The major difference is that F# is intended to be a general-purpose programming language and is multi-paradigm since it also supports imperative and OOP programming, whereas Wake is intended to be a programming language to support a build system and does not support the imperative or OOP programming paradigms. However, while more simple, the Wake programming language is rather powerful.
```

Reply:
I hadn't reviewed that since I wrote it I think, and so I removed/reworded a large portion of the middle, as it's pretty much irrelevant to a doc like this I think. Wake should already be sold by this point. :) [Updated here.](https://github.com/sifive/wake/pull/1119/commits/661f3f971e4c9acae7e6d4d0a8592de338190cc2)